tests/integration/test_time_manager.py

```diff
@@ -0,0 +1,191 @@
+"""
+This module contains tests that are devised to guarantee a correct integration between of the
+TimeManager class into the PorePy models.
+
+Brief summary of tests:
+    * test_default_behaviour: Tests default behaviour of the TimeManager in all models. This
+        also includes static models.
+    * test_constant_dt: Tests time management when constant time step is used on all
+        time-dependent models.
+    * test_linear_non_constant_dt: Tests time management on time-dependent linear models when
+        non-constant time step is used.
+    * test_time_step_adaptation_after_convergence: Tests time adaptation on non-linear
+        time-dependent models where convergence is achieved in all time steps.
+    * test_time_step_adapation_after_failure: Tests time adaptation on non-linear
+        time-dependent models where recomputation-after-failed-to-converge is necessary in at
+        least one time level.
+
+"""
+
+import numpy as np
+import pytest
+
+import porepy as pp
+from tests.common.slightly_compressible_flow_examples import NonLinearSCF
+
+# Model classes to be tested
+standard_models = [
+    pp.IncompressibleFlow,
+    pp.SlightlyCompressibleFlow,
+    pp.ContactMechanics,
+    pp.ContactMechanicsBiot,
+    pp.THM,
+]
+
+time_dependent_models = [pp.SlightlyCompressibleFlow, pp.ContactMechanicsBiot, pp.THM]
+
+linear_time_dependent_models = [
+    pp.SlightlyCompressibleFlow,
+    pp.ContactMechanicsBiot,
+    pp.THM,
+]
+
+nonlinear_time_dependent_models = [NonLinearSCF]
+
+
+# -----> Testing default behaviour
+@pytest.mark.parametrize("model_class", standard_models)
+def test_default_behaviour(model_class):
+    """Test the default behaviour of the models"""
+
+    model = model_class(params={"use_ad": True})
+
+    if not model._is_time_dependent():
+        pp.run_stationary_model(model, model.params)
+        assert not hasattr(model, "time_manager")
+    else:
+        pp.run_time_dependent_model(model, model.params)
+        assert model.time_manager.is_constant
+        assert model.time_manager.time_init == 0
+        assert model.time_manager.time_final == 1
+        assert model.time_manager.time == 1
+        assert model.time_manager.dt == 1
+        assert model._ad.time_step.evaluate(model.dof_manager) == model.time_manager.dt
+
+
+# -----> Testing constant time step
+tm0 = pp.TimeManager([0, 1, 2], 1, constant_dt=True)
+tm1 = pp.TimeManager([0, 0.01, 0.02, 0.05], 0.01, constant_dt=True)
+tm2 = pp.TimeManager([0, pp.HOUR, 2 * pp.HOUR, 10 * pp.HOUR], pp.HOUR, constant_dt=True)
+
+
+@pytest.mark.parametrize("model_class", time_dependent_models)
+@pytest.mark.parametrize("time_manager", [tm0, tm1, tm2])
+def test_constant_dt(request, model_class, time_manager):
+    """Test models when constant time step is used"""
```

Comment:
Am I right that this is basically a test of TimeManager only, not its integration with models?